### ISSUE TITLE
Hi, could you please tell me how to do the pre-training you mentioned?

### ISSUE THREAD

**Issue by starrism-yu** [kumar10725/Keras-Determinant-based-Mutual-Information-loss-DMI-loss-#1]
hi, Could you please tell me how to do the pre-training you mentioned? Thanks

**Comment by kumar10725** [comment]
You can train the model using a different loss function like crossentropy or mse. Then you can load those weights and train again using peerloss.

**Comment by starrism-yu** [comment]
Thanks, I would also like to ask about PeerLoss. There are two pixels involved in the paper, and the formula is Lpeer = L(F (Xn),Yn) -L (F (Xn1)-Yn2). However, it seems that there is no corresponding here in your code. Compared with DMI Loss, it only seems to increase a coefficient. Can you explain it, please

**Comment by kumar10725** [comment]
I think what you are referring to is here:
`U = ((1/tf.cast(tf.keras.backend.shape(y_true)[0], tf.float32))*tf.matmul(tf.transpose(y_pred), y_true))`
there is indeed two pixels here in `y_pred` and `y_true`.

**Comment by starrism-yu** [comment]
Do you mean y_pred and y_true are two dimensions? That is, the input data of this function is two points, instead of one batch size data?

**Comment by kumar10725** [comment]
I think there might be some misunderstanding from the paper. 
The input data for this function is batch that makes the input two dimension, (N,1), where N is the batch size. That way the transpose function can be applied to points. In tensorflow loss function's input is always batch.
